### PR TITLE
signer/core: fix reference issue in key derivation

### PR DESCRIPTION
**Pull request by gballet**
Currently in ~~`clé`~~ `clef`, the derivation process reuses the same derivation path array as input to the `Derive` function. That `Derive` function creates an account and keeps a pointer to the derivation path for other purposes.

Since the derivation path gets updated each time `Derive` is called, all accounts end up having the rug pulled from under their feet: the derivation path they point to will be that of the first non-valid account.

So this PR simply makes sure that one derivation path is allocated each time.

### CODE REVIEW

**Comment by karalabe** [comment]
This seems like a bug in a specific derivation method. It should definitely not keep a pointer to the path, rather make a copy if it needs it. Where does it originate from though, Ledger and Trezor only use it for a request, but don't mutate it as far as I see it.

**Comment by karalabe** [comment]
Ah, indeed it does keep a reference. Please make a copy here: https://github.com/ethereum/go-ethereum/blob/master/accounts/usbwallet/wallet.go#L482.

File: accounts/usbwallet/wallet.go [hunk 1]
```diff
@@ -479,7 +479,7 @@ func (w *wallet) Derive(path accounts.DerivationPath, pin bool) (accounts.Accoun
 
 	if _, ok := w.paths[address]; !ok {
 		w.accounts = append(w.accounts, account)
-		w.paths[address] = path
+		copy(w.paths[address], path)
```
Comment (karalabe) [inline accounts/usbwallet/wallet.go]: This doesn't make a new copy. It retrieves `w.paths[address]`, which will be an empty slice (nil) and then copy into it, limited to the length of the smallest operand (i.e. `nil`, length of 0`) and return. I.e. this is a noop.

Comment (gballet) [reply accounts/usbwallet/wallet.go]: yeah I just saw debugging the other PR :disappointed:

Comment (karalabe) [reply accounts/usbwallet/wallet.go]: Alternatively, we can make a `Copy` method for `accounts.DerivationPath`, might come in handy long term for Clef too.

**Pull request merged**